backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -108,13 +122,17 @@ export class SegmentService {
     });
   }
 
-  public async getSingleSegmentWithStatus(segmentId: string, logger: UpgradeLogger): Promise<Segment> {
+  public async getSingleSegmentWithStatus(segmentId: string, logger: UpgradeLogger): Promise<SegmentWithStatus> {
+    const allSegmentData = await this.getAllSegmentsAndSubsegments(logger);
     const segmentData = await this.getSegmentById(segmentId, logger);
-    return (await this.getSegmentStatus([segmentData])).segmentsData[0];
+    const segmentWithStatus = (await this.getSegmentStatus(allSegmentData)).segmentsData.find(
+      (segment: Segment) => segment.id === segmentId
+    );
+    return { ...segmentData, status: segmentWithStatus.status };
   }
 
   public async getAllSegmentWithStatus(logger: UpgradeLogger): Promise<getSegmentData> {
```

Comment:
Now getSegmentStatus will always need all the segment data that should be part of this function as well.

Reply:
Yes, we will need that.